### PR TITLE
filter_log_to_metrics: Add interval options for metric emission

### PR DESCRIPTION
**Pull request by drbugfinder-work**

### CODE REVIEW

File: pipeline/filters/log_to_metrics.md [hunk 1]
```diff
@@ -209,4 +211,4 @@ As you can see in the output, there are per default the buckets `0.005, 0.01, 0.
 Please note, that the `+Inf` bucket will always be included implicitly. The buckets in a histogram are cumulative, so a value added to one bucket will add to all larger buckets, too.
 
 
-You can also see, that all the kubernetes labels have been attached to the metric, idential to the behavior of `label_field` described in [the previous chapter](#metric-label_values). That results in two sets for the histogram.
\ No newline at end of file
+You can also see, that all the kubernetes labels have been attached to the metric, idential to the behavior of `label_field` described in [the previous chapter](#metric-label_values). That results in two sets for the histogram.
```
Comment (alexakreizinger) [inline pipeline/filters/log_to_metrics.md:214]:
```suggestion
This filter also attaches Kubernetes labels to each metric, identical to the behavior of `label_field`.
This results in two sets for the histogram.
```

**Review by alexakreizinger — COMMENTED** [review]
thanks for submitting a PR, @drbugfinder-work! I've made a few suggestions in accordance with our style guide, and then there's just one question I wanted to clarify about the behavior of `Interval_Sec` vs `Interval_Nsec` 🙂

File: pipeline/filters/log_to_metrics.md [hunk 2]
```diff
@@ -28,6 +28,8 @@ The plugin supports the following configuration parameters:
 | kubernetes_mode |  If enabled, it will automatically put pod_id, pod_name, namespace_name, docker_id and container_name into the metric as labels. This option is intended to be used in combination with the [kubernetes](./kubernetes.md) filter plugin, which fills those fields. | | 
 | Regex | Include records in which the content of KEY matches the regular expression. | |  KEY  REGEX 
 | Exclude | Exclude records in which the content of KEY matches the regular expression. | |  KEY  REGEX 
+| Interval\_Sec | Defines interval in seconds for metrics emission. Default value is _0_. If Interval\_Sec and Interval\_Nsec are not set or both have value _0_, metrics are emitted immediately into the pipeline for every filter match. Otherwise new metric updates will only be emitted into the pipeline in the defined interval. Setting an interval can lower the resource consumption in high load situations. |
+| Interval\_Nsec | Nanoseconds interval for metrics emission. It works in conjunction with the Interval\_Sec configuration key. Default value is _0_. |
```
[the remark this answers is not in the document]
Comment (drbugfinder-work) [reply pipeline/filters/log_to_metrics.md:32]: Hi @alexakreizinger!
Thank you for the rework. I wasn't aware of any style guide for the docs. Regarding your question about the two interval settings - it's exactly the same as with many other plugins (e.g. [health](https://docs.fluentbit.io/manual/pipeline/inputs/health), [process](https://docs.fluentbit.io/manual/pipeline/inputs/process), [disk-io-metrics](https://docs.fluentbit.io/manual/pipeline/inputs/disk-io-metrics), [cpu-metrics](https://docs.fluentbit.io/manual/pipeline/inputs/cpu-metrics), ...). The one parameter allows you to set the seconds (as integers) and the other allows you to set nanoseconds. Both are summed up in the code, so you can for example set seconds to 5 and nanoseconds to 500.000.000 to get 5,5 seconds.

[the remark this answers is not in the document]
Comment (alexakreizinger) [reply pipeline/filters/log_to_metrics.md]: no worries about the style guide—we don't have anything publicly available that contributors are expected to follow, so we're happy to make suggestions/adjustments accordingly. 

and that makes sense; thanks for clarifying!

**Review by alexakreizinger — APPROVED** [review]

**Pull request merged**